freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -143,13 +149,30 @@ private String serviceAdd(String serviceName, String hostname, String realm, Fre
         RPCResponse<com.sequenceiq.freeipa.client.model.Service> response;
         com.sequenceiq.freeipa.client.model.Service service;
         try {
-            response = ipaClient.addService(serviceName + "/" + hostname + "@" + realm);
-            if (response == null || response.getResult() == null) {
-                return null;
+            String canonicalPrincipal = serviceName + "/" + hostname + "@" + realm;
+            try {
+                response = ipaClient.addService(canonicalPrincipal);
+                if (response == null || response.getResult() == null) {
+                    return null;
+                }
+                service = (com.sequenceiq.freeipa.client.model.Service) response.getResult();
+            } catch (FreeIpaClientException e) {
+                if (!Optional.ofNullable(e.getCause())
```

Comment:
this long if seems the same to me, could you refactor it to a method so we don't duplicate it?

Reply:
I refactored the logic so its not duplicated.